### PR TITLE
NP-2363 unified logging

### PR DESCRIPTION
**Pull request by Carmendelope**
#### What does this PR do?
- Implements unified-logging.Search
- Add an instances cache to avoid accessing the database to ask for reduced summary instances more times than necessary
#### What is missing?
LogResponse has an `is_dead` field that is not filled for now. To fill in this field, we need the catalog to be implemented

#### What are the relevant tickets?

- [NP-2363](https://nalej.atlassian.net/browse/NP-2363)

#### Questions

### CODE REVIEW

File: internal/pkg/server/unified_logging/manager.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+		From:           request.From,
+		To:             request.To,
+		Entries:        logResponse,
+	}, nil
+}
+
+// getNamesFromSummary returns the name of the serviceGroupId and the serviceId
+func (m *Manager) getNamesFromSummary(serviceGroupId string, serviceId string, inst *grpc_application_go.AppInstanceReducedSummary) (string, string) {
+
+	groupName := ""
```
Comment (dhiguero) [inline internal/pkg/server/unified_logging/manager.go]:
```suggestion
	groupName := "Unknown"
```

File: internal/pkg/server/unified_logging/manager.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	serviceName := ""
```
Comment (dhiguero) [inline internal/pkg/server/unified_logging/manager.go]:
```suggestion
	serviceName := "Unknown"
```

File: internal/pkg/server/unified_logging/manager.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+	}
+	if logEntry.AppInstanceId == "" {
+		log.Warn().Msg("unable to expand log information, app_instance_id is empty")
+		return logEntry
+	}
+
+	summary, err := m.instHelper.RetrieveInstanceSummary(organizationId, logEntry.AppInstanceId)
+	if err != nil {
+		log.Warn().Interface("trace", err.StackTrace()).Str("organizationId", organizationId).Str("appInstanceId", logEntry.AppInstanceId).Msg("error getting reduced summary")
+		return logEntry
```
Comment (dhiguero) [inline internal/pkg/server/unified_logging/manager.go]:
```suggestion
                logEntry.AppDescriptorName = "Unknown"
                logEntry.ServiceGroupName = "Unknown"
 	        logEntry.ServiceName = "Unknown"
		return logEntry
```

File: internal/pkg/utils/instances_helper.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	defer cancel()
+	retrievedSummary, err := i.appClient.GetAppInstanceReducedSummary(ctx, &grpc_application_go.AppInstanceId{
+		OrganizationId: organizationId,
+		AppInstanceId:  appInstanceId,
+	})
+	if err != nil {
+		return nil, conversions.ToDerror(err)
+	}
+
+	_ = i.cache.Add(pk, summary)
```
Comment (dhiguero) [inline internal/pkg/utils/instances_helper.go]:
```suggestion
	_ = i.cache.Add(pk, retrievedSummary)
```

File: internal/pkg/server/unified_logging/manager.go [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+// Manager structure with the required clients for roles operations.
+type Manager struct {
+	unifiedLogging grpc_unified_logging_go.CoordinatorClient
+	appClient      grpc_application_go.ApplicationsClient
+	instHelper     *utils.InstancesHelper
+}
+
+// NewManager creates a Manager using a set of clients.
+func NewManager(unifiedLogging grpc_unified_logging_go.CoordinatorClient, appClient grpc_application_go.ApplicationsClient) Manager {
+	helper, _ := utils.NewInstancesHelper(appClient, DefaultCacheEntries)
```
Comment (dhiguero) [inline internal/pkg/server/unified_logging/manager.go]: if the helper cannot be built, we should fail the creation of the manager, and the service itself should fail.

File: internal/pkg/server/unified_logging/manager.go (hunk 1 above)
Comment (Carmendelope) [reply internal/pkg/server/unified_logging/manager.go]: Done

File: internal/pkg/server/unified_logging/manager.go (hunk 2 above)
Comment (Carmendelope) [reply internal/pkg/server/unified_logging/manager.go]: done

File: internal/pkg/server/unified_logging/manager.go (hunk 3 above)
Comment (Carmendelope) [reply internal/pkg/server/unified_logging/manager.go]: Done

File: internal/pkg/utils/instances_helper.go (hunk 4 above)
Comment (Carmendelope) [reply internal/pkg/utils/instances_helper.go]: done

File: internal/pkg/server/unified_logging/manager.go (hunk 5 above)
Comment (Carmendelope) [reply internal/pkg/server/unified_logging/manager.go]: Done

**Pull request merged**